File: frontend/src/lib/inventory.ts
// 库存(订单履约跟踪)前端类型 + API。对齐后端 stock_balance_service.compute_stock_balance。
// 🔴 纯派生,零成本/供应商字段 —— 读投影天然无红线,零脱敏分支(契约 §3)。
import { api } from "./api";
import type { Page } from "./catalog";
import { qs } from "./qs";

/** `/api/v1/inventory` 列表行(契约 §3 逐字字段)。 */
export interface StockBalanceItem {
  sales_order_id: number;
  sales_order_no: string;
  sku_code: string;
  name: string;
  spec_text: string;
  unit: string;
  ordered_qty: number | string;
  inbound_qty: number | string;
  outbound_qty: number | string;
  available_qty: number | string;
}

/** SO 详情内嵌块(契约 §3:该 SO 各 SKU 四量,含已入 0 行)。附 sku_id 供行 key/未来下钻。 */
export interface StockBalanceLine {
  sku_id: number;
  sku_code: string;
  name: string;
  spec_text: string;
  unit: string;
  ordered_qty: number | string;
  inbound_qty: number | string;
  outbound_qty: number | string;
  available_qty: number | string;
}

export interface InventoryListFilters {
  sales_order_id?: number;
  sku_id?: number;
  q?: string;
  /** 默认口径省略 = available_qty>0(在库);传 "history" = 加显已发完行(履约史 inbound>0 OR outbound>0;未入库行不进本页,契约 §2/§5)。 */
  scope?: "history";
  page?: number;
  size?: number;
}

export const inventoryApi = {
  list: (p: InventoryListFilters) =>
    api.get<Page<StockBalanceItem>>(`/api/v1/inventory${qs(p as Record<string, unknown>)}`),
};

File: backend/app/schemas/inventory.py
"""库存(订单履约跟踪)读投影。纯派生四量 + 展示三件套(SKU 当前档)。
无成本/供应商/金额字段 → 零红线、零脱敏分支。"""
from __future__ import annotations

from pydantic import BaseModel


class StockBalanceRow(BaseModel):
    """每 (销售单, SKU) 一行:订购/已入库/已出库(本步恒 0)/可发。
    展示字段(品名/规格串/单位)取 SKU 当前档,按所属 SO 语言渲染。"""
    sales_order_id: int
    sales_order_no: str
    sku_id: int
    sku_code: str
    name: str
    spec_text: str
    unit: str
    ordered_qty: float
    inbound_qty: float
    outbound_qty: float
    disposition_qty: float
    available_qty: float
